src/fetchExceptions.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  Response as FetchResponse
+} from 'node-fetch'
+
+
+export const checkStatusCode = (response: FetchResponse): number | undefined => {
+  const { hostname } = new URL(response.url)
+  const code = exceptionMap.get(hostname)
+  if (code === response.status) {
+    console.log(`Returning status code 418 for ${hostname}`)
+    return 418
+  }
+}
+
+/**
+ * Known response codes to override with teapot response
+ */
+const exceptionMap = new Map([
+  ['api.binance.org', 403]
+])
```

Comment:
We don't know if all requests are going to be only requiring a check for the status code. Some requests might need to check the status code and perhaps the body, or maybe even the header. One idea is to implement a function for each domain with the signature `(response: FetchResponse) => boolean`, and map the domain to that function. 

We don't need to return `number` for `checkStatusCode` because we're only ever going to return `418` if necessary, otherwise return the response as is. A name like `checkIsRateLimitResponse` is more self-documenting to what we're doing here.

Reply:
OTS result for a future PR `checkCanNotHandleRequest` or similar. This will return a boolean indicating whether the CORS proxy server should return a teapot error.